andela/travel_tool_front
# #162373925 Fix dashboard page to show correct error message to user

#### What does this PR do?
Display error messages to a user viewing dashboard page

#### Description of Task to be completed?
- get a server error response
- display error response on each dashboard component accordingly

#### How should this be manually tested?
- Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Switch to the project directory
   `cd travel_tool_front`
 - Checkout the **bug-dashboard-error-messages-162373925** branch
   `git checkout bug-dashboard-error-messages-162373925`
 - Install dependencies and start the server
   - `make start` *if you have docker*
   - `yarn install` then `yarn start` *if you do not have docker*
 - Start the [backend](https://github.com/andela/travel_tool_back) of this project.
 - Navigate to [travela](http://travela-local.andela.com:3000/) in your browser and log in with your Andela account. 
 - Login as admin
- Navigate to [travela requests](http://travela-local.andela.com:3000/dashboard) in your browser.
 - Create a travel request
 - You should see travel requests
 - Open the back end module, pick an endpoint that returns data displayed on the dashboard, for instance,**get travel readiness**, 
```src/modules/analytics/index.js``` 
comment out this route `/analytics/readiness`
- Restart the server
- Reload the dashboard
- You should see that travel readiness component only has an error message

#### Any background context you want to provide?


#### What are the relevant pivotal tracker stories?
[#162373925](https://www.pivotaltracker.com/story/show/162373925, "show error on dashboard page")

#### Screenshots (if appropriate)
Previously it shows no data found when there's server error
<img width="1680" alt="screen shot 2018-12-12 at 9 13 02 am" src="https://user-images.githubusercontent.com/24890605/49877518-de341380-fe25-11e8-8ea0-9da80a7cc5a1.png">

Current It shows an error message to the particular component when there's a server error
<img width="1680" alt="screen shot 2018-12-13 at 5 57 36 pm" src="https://user-images.githubusercontent.com/24890605/49956105-b798da00-ff04-11e8-9257-044219c0d290.png">

#### Questions:
